recipes/autoconf/all/conanfile.py

```diff
@@ -1,122 +1,146 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-import contextlib
-import os
+from conan import ConanFile
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rmdir, apply_conandata_patches, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import unix_path, is_msvc
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.51.3"
 
 
 class AutoconfConan(ConanFile):
     name = "autoconf"
+    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.gnu.org/software/autoconf/"
-    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
     topics = ("autoconf", "configure", "build")
-    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     settings = "os", "arch", "compiler", "build_type"
 
-    exports_sources = "patches/*"
-
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
-
     @property
     def _settings_build(self):
+        # TODO: Remove for Conan v2
         return getattr(self, "settings_build", self.settings)
 
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    def configure(self):
+        self.win_bash = self._settings_build.os == "Windows"
+
+    def layout(self):
+        basic_layout(self, src_folder="autoconf")
+
     def requirements(self):
         self.requires("m4/1.4.19")
 
-    def build_requirements(self):
-        if hasattr(self, "settings_build"):
-            self.build_requires("m4/1.4.19")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def build_requirements(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```

Comment:
Nice, as msys2 is already ported, we no longer need CONAN_BASH_PATH

Reply:
Credit where credit is due, stole this from one of @SpaceIm PRs